cmd/proxygenerator/interceptor.go

```diff
@@ -109,39 +109,65 @@ func NewPayloadVisitorInterceptor(options PayloadVisitorInterceptorOptions) (grp
 
 		err := invoker(ctx, method, req, response, cc, opts...)
 		if err != nil {
-			if options.Inbound != nil {
-				stat, ok := status.FromError(err)
-				if ok {
-					// user provided payloads can sometimes end up in the status details of
-					// gRPC errors, make sure to visit those as well
-					newStatus := status.New(stat.Code(), stat.Message())
-					var newDetails []protoadapt.MessageV1
-					for _, detail := range stat.Details() {
-						detailAny, ok := detail.(*anypb.Any)
-						if ok && ({{ range $i, $name := .GrpcPayload }}{{ if $i }} || {{ end }}detailAny.MessageName() == "{{$name}}"{{ end }}) {
-							err = VisitPayloads(ctx, detailAny, *options.Inbound)
-							if err != nil {
-								return err
-							}
+			return parseGrpcPayload(ctx, err, options.Inbound)
+		}
+
+		if resMsg, ok := response.(proto.Message); ok && options.Inbound != nil {
+			return VisitPayloads(ctx, resMsg, *options.Inbound)	
+		}
+		
+		return nil
+	}, nil
+}
+
+func parseGrpcPayload(ctx context.Context, err error, inbound *VisitPayloadsOptions) error {
+	if inbound != nil {
+		stat, ok := status.FromError(err)
+		if ok {
+			// user provided payloads can sometimes end up in the status details of
+			// gRPC errors, make sure to visit those as well
+			newStatus := status.New(stat.Code(), stat.Message())
+			var newDetails []protoadapt.MessageV1
+			for _, detail := range stat.Details() {
+				detailAny, okAny := detail.(*anypb.Any)
```

Comment:
This can be done inline in the if, e.g. `if msg, ok := detail.(*anypb.Any); ok {` if it's easier for you